src/main/controllers/DeleteCourtSubscriptionConfirmationController.ts

```diff
@@ -0,0 +1,48 @@
+import { PipRequest } from '../models/request/PipRequest';
+import { Response } from 'express';
+import { cloneDeep } from 'lodash';
+import { LocationService } from '../service/locationService';
+import { SubscriptionService } from '../service/subscriptionService';
+import { UserManagementService } from '../service/userManagementService';
+
+const locationService = new LocationService();
+const subscriptionsService = new SubscriptionService();
+const userManagementService = new UserManagementService();
+
+export default class DeleteCourtSubscriptionConfirmationController {
+    public async post(req: PipRequest, res: Response): Promise<void> {
+        const formData = req.body;
+        const court = await locationService.getLocationById(formData.locationId as unknown as number);
+        if (formData['delete-choice'] == 'yes') {
+            const response = await subscriptionsService.deleteLocationSubscription(
+                formData.locationId,
+                req.user?.['provenanceUserId']
+            );
+            if (response === null) {
+                res.render('delete-court-subscription-confirmation', {
+                    ...cloneDeep(req.i18n.getDataByLanguage(req.lng)['delete-court-subscription-confirmation']),
+                    court: locationService.formatCourtValue(court),
+                    apiError: true,
+                    errorMessage: 'Unknown error when attempting to delete all the subscription for the court',
```

Comment:
I think in the error message it should be 'subscriptions'

Reply:
fixed.